lang/py/src/avro/schema.py

```diff
@@ -551,6 +506,69 @@ def to_json(self, names=None):
   def __eq__(self, that):
     return self.props == that.props
 
+#
+# Decimal Fixed Type
+#
+
+class FixedDecimalSchema(FixedSchema, LogicalSchema):
+  def __init__(self, schema_type, size, name, precision, scale=0, namespace=None, names=None, other_props=None):
+    if schema_type != 'fixed':
+      raise SchemaParseException('Logical type DECIMAL must be backed by fixed or bytes.')
+    if size < 0:
+      raise SchemaParseException('Size of a fixed length should be greater than 0')
+
+    self._validate({'max_precision': self._max_precision(size),
+                   'precision': precision,
+                   'scale': scale,
+                   'size': size})
+
+    LogicalSchema.__init__(self, 'decimal')
+    FixedSchema.__init__(self, name, namespace, size, names, other_props)
+    self.set_prop('precision', precision)
+    self.set_prop('scale', scale)
+
+  def _validate(self, validation_props=None):
```

Comment:
The validation to be consolidated are the ones for decimal logical types, including:
1. precision must be a positive integer
2. scale must be 0 or a positive integer
3. scale must be less than precision
4. precision must be less than max precision allowed.

If the spec of decimal logical type is changed, both places will need to change at the same time. This is to avoid the duplicate work.

Also, it doesn't require public functions to do that. My pb is one way to do it. I think there could be other ways to do so. Even you decide to duplicate the code in both places, at least to make the validation the same. Right now it doesn't seem they are.